src/server/list_family_test.cc

```diff
@@ -700,6 +700,22 @@ TEST_F(ListFamilyTest, BRPopLPushSingleShardBug2857) {
   EXPECT_THAT(resp, ArgType(RespExpr::NIL_ARRAY));
 }
 
+TEST_F(ListFamilyTest, BRPopLPushSingleShardBug4569) {
```

Comment:
I was playing around and this test passes `before` and `after` your changes. 

Similarly, the following also passes with and without your changes:

```
> blmove foo bar left left 0
> blocker here


another client 
> lpush foo bar 
```

Works without any issues with `--proactor_threads` to enforce a single shard.

Reply:
Also something irrelevant but surprising you can do:

```
blmove foo bar left left 0

set foo bar
```

blmove will never wake up. We don't care though, this is the standard behaviour for Valkey (I checked)

Reply:
@romange Then we should add that here as well so we can actually test the fix :smile:

Reply:
we dont need to put cluster_mode, I discovered the bug originaly on cluster emulated but this is just because the flow was running the rpoplpush on single shard.
@kostasrim what did you try to test on your cli?
using flag --proactor_threads=1
running :
brpop x 0

lpush y val
rpoplpush y x

this does not wakes up the blocking command without the fix

Reply:
also I now checked this test again without my fix and this test is blocked on fb0.Join();

Reply:
I looked on my redis-cli history, I was doing the wrong command. It reproduced now